src/cdk/scrolling/virtual-for-of.ts

```diff
@@ -129,21 +138,27 @@ export class CdkVirtualForOf<T> implements
    * The size of the cache used to store templates that are not being used for re-use later.
    * Setting the cache size to `0` will disable caching. Defaults to 20 templates.
    */
-  @Input() cdkVirtualForTemplateCacheSize: number = 20;
+  @Input()
+  get cdkVirtualForTemplateCacheSize() {
+    return this._viewRepeater.viewCacheSize;
+  }
+  set cdkVirtualForTemplateCacheSize(size: number | string) {
```

Comment:
This should just be `number`, and you can add this to the bottom of this class to tell Angular to accept other data types:
```ts
static ngAcceptInputType_cdkVirtualForTemplateCacheSize: NumberInput;
```

Reply:
Done